### PR TITLE
ci: Use a new release action

### PR DESCRIPTION
**Pull request by eiffel-fl**
Hi.

This PR supersedes #1068 by using [softprops/action-gh-release](https://github.com/softprops/action-gh-release) instead of [actions/create-release](https://github.com/actions/create-release) which is now archived.
It also appends the tag name to release artifacts.
It was tested and validated in a [fork](https://github.com/eiffel-fl/inspektor-gadget/releases/tag/vtest42).

Best regards.

### CODE REVIEW

**Comment by blanquicet** [comment]
Thanks for working on this!

> It was tested and validated in a [fork](https://github.com/eiffel-fl/inspektor-gadget/releases/tag/vtest42).
>
Why are the local-gadget binaries not present in the release assets?

**Comment by blanquicet** [comment]
> > Why are the local-gadget binaries not present in the release assets?
> 
> To speed up the whole thing, I [removed some jobs](https://github.com/eiffel-fl/inspektor-gadget/commit/8a96066593c375a27f29b31d8f15430ca6e77bb3) from the `needs` of the `release` job. So, at the time where the `release` job was run, the `local-gadget` artifacts were not available.
>
Oh, I just realized that the release job doesn't have `test-integration-local-gadget` as a need. I am wondering why not 🤔 ...

**Comment by mqasimsarfraz** [comment]
> Oh, I just realized that the release job doesn't have `test-integration-local-gadget` as a need. I am wondering why not thinking ...

I thought we leave the new GH jobs out of release by choice to test the initial stability. Probably about time to should add  `test-integration-local-gadget` to release as well.

File: .github/workflows/inspektor-gadget.yml [hunk 1]
```diff
@@ -686,20 +687,22 @@ jobs:
...
+      shell: bash
+      run: |
+        for i in *-gadget-*-*-tar-gz/*-gadget-*-*.tar.gz; do
+          mv $i $(dirname $i)/$(basename $i .tar.gz)-${GITHUB_REF_NAME}.tar.gz
+        done
+    - name: Upload Gadget Release *.tar.gz
       uses: csexton/release-asset-action@v2
       with:
-        pattern: "*-gadget-*-*-tar-gz/*-gadget-*-*.tar.gz"
+        pattern: "*-gadget-*-*-tar-gz/*.tar.gz"
```
Comment (blanquicet) [inline .github/workflows/inspektor-gadget.yml:705]: Why is this change needed?

File: .github/workflows/inspektor-gadget.yml [hunk 2]
```diff
@@ -686,20 +687,22 @@ jobs:
...
+        name: Release ${{ github.ref }}
     - name: Get all artifacts.
       uses: actions/download-artifact@v2
-    - name: Upload Gadget Release *-gadget-*-*.tar.gz
+    - name: Rename all artifacts to *-${{ env.GITHUB_REF_NAME }}.tar.gz
+      shell: bash
+      run: |
+        for i in *-gadget-*-*-tar-gz/*-gadget-*-*.tar.gz; do
+          mv $i $(dirname $i)/$(basename $i .tar.gz)-${GITHUB_REF_NAME}.tar.gz
+        done
```
Comment (blanquicet) [inline .github/workflows/inspektor-gadget.yml:701]: Could you please verify this step is working properly for the local-gadget binaries? Maybe just adding `build-local-gadget` as a dependency for the `release` job (just for testing).

**Comment by blanquicet** [comment]
@mqasimsarfraz after these two weeks, I would say that it is working properly and can be added as a dependency. @eiffel-fl, of course, this discussion about local-gadget doesn't block this PR, it can be managed in another PR.

File: .github/workflows/inspektor-gadget.yml (hunk 2 above)
Comment (blanquicet) [reply .github/workflows/inspektor-gadget.yml:701]: Thanks!

**Review by blanquicet — APPROVED** [review]
LGTM!
